app/uk/gov/hmrc/gform/models/DataRetrieveAll.scala

```diff
@@ -30,9 +31,10 @@ final case class DataRetrieveAll(lookup: Map[DataRetrieveId, DataRetrieve]) exte
 object DataRetrieveAll {
   val empty: DataRetrieveAll = DataRetrieveAll(Map.empty)
 
-  def from[A <: PageMode](formModel: FormModel[A]): DataRetrieveAll =
+  def from[A <: PageMode](formModel: FormModel[A], dataRetrieve: Option[NonEmptyList[DataRetrieve]]): DataRetrieveAll =
```

Comment:
why do you need to add additional argument here? dataRetrieve is in formModel already `formModel.dataRetrieve`

Reply:
My IDE shows that the only usage of DataRetrieveAll is from here:
https://github.com/hmrc/gform-frontend/blob/2b74705f47cd691cfd851fcc72a83e59bdddf739/app/uk/gov/hmrc/gform/models/FormModel.scala#L155
it takes `this` and `dataRetrieve`, from from what I understand `this.dataRetrieve` is the same as `dataRetrieve` in this context.

Reply:
Fixed